lib/Controller/SettingsController.php

```diff
@@ -87,37 +87,47 @@ public function save($avMode, $avSocket, $avHost, $avPort,
 		$avCmdOptions, $avPath, $avInfectedAction, $avStreamMaxLength,
 						 $avMaxFileSize, $avRequestService, $avResponseHeader
 	) {
-		$this->settings->setAvMode($avMode);
-		if ($avMode === 'executable') {
-			$this->settings->setAvCmdOptions($avCmdOptions);
-			$this->settings->setAvPath($avPath);
-		} elseif ($avMode === 'daemon') {
-			$this->settings->setAvPort($avPort);
-			$this->settings->setAvHost($avHost);
-		} elseif ($avMode === 'socket') {
-			$this->settings->setAvSocket($avSocket);
-		} elseif ($avMode === 'icap') {
-			$this->settings->setAvPort($avPort);
-			$this->settings->setAvHost($avHost);
-			$this->settings->setAvRequestService($avRequestService);
-			$this->settings->setAvResponseHeader($avResponseHeader);
-		}
-
-		$this->settings->setAvInfectedAction($avInfectedAction);
-		$this->settings->setAvStreamMaxLength($avStreamMaxLength);
-		$this->settings->setAvMaxFileSize($avMaxFileSize);
+		try {
+			if ($avMode === 'executable') {
+				$this->settings->setAvCmdOptions($avCmdOptions);
+				$this->settings->setAvPath($avPath);
+			} elseif ($avMode === 'daemon') {
+				$this->settings->setAvPort($avPort);
+				$this->settings->setAvHost($avHost);
+			} elseif ($avMode === 'socket') {
+				$this->settings->setAvSocket($avSocket);
+			} elseif ($avMode === 'icap') {
+				$this->settings->setAvPort($avPort);
+				$this->settings->setAvHost($avHost);
+				$this->settings->setAvRequestService($avRequestService);
+				$this->settings->setAvResponseHeader($avResponseHeader);
+			}
 
-		$connectionStatus = (int)$this->scannerFactory->testConnection($this->settings);
+			$this->settings->setAvInfectedAction($avInfectedAction);
+			$this->settings->setAvStreamMaxLength($avStreamMaxLength);
+			$this->settings->setAvMaxFileSize($avMaxFileSize);
+			$this->settings->setAvMode($avMode);
 
-		return new JSONResponse(
-			['data' =>
-				['message' =>
-					(string) $this->l10n->t('Saved')
+			$connectionStatus = (int)$this->scannerFactory->testConnection($this->settings);
+			$response = [
+				'data' => [
+					'message' => (string) $this->l10n->t('Saved')
 				],
 				'connection' => $connectionStatus,
 				'status' => 'success',
 				'settings' => $this->settings->getAllValues()
-			]
-		);
+			];
+		} catch (\UnexpectedValueException $e) {
+			$response = [
+				'data' => [
+					'message' => $e->getMessage()
+				],
+				'connection' => 1,
```

Comment:
I'm worried this "1" value isn't documented anywhere...
Could I change it to "-1" or "999"? Does it have any meaning (such as 1 = cannot connect, 2 = test file failed to be detected, etc)

Reply:
unrelated to the the issue